### ISSUE TITLE
[COMPONENT] Create Divider/Grouping component for list

### ISSUE THREAD

**Issue by MadsBuchmann** [kirbydesign/designsystem#1886]
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the solution you'd like
A component that allows for dividing a kirby-list into groups should be created. 
With the current "prop"-based (for lack of a better word) list the `[showDivider]` input property is used. 

With the new declarative approach to the list it should be possible to use it along the lines of: 
```html
<h1>Items!</h1>
<kirby-list>
   <kirby-item-group> 
      <h2>1. Items</h2>
      <kirby-item>Item 1.1</kirby-item>
      <kirby-item>Item 1.2</kirby-item>
      <kirby-item>Item 1.3</kirby-item>
   </kirby-item-group> 
   <kirby-item-group> 
      <h2>2. more Items</h2>
      <kirby-item>Item 2.1</kirby-item>
      <kirby-item>Item 2.2</kirby-item>
      <kirby-item>Item 2.3</kirby-item>
   </kirby-item-group> 
</kirby-list>
```

or perhaps: 
```html 
<h1>Items!</h1>
<kirby-list>
      <kirby-item-divider> 
          1. Items
      </kirby-item-divider>
      <kirby-item>Item 1.1</kirby-item>
      <kirby-item>Item 1.2</kirby-item>
      <kirby-item>Item 1.3</kirby-item>
      <kirby-item-divider>  
          2. More Items
      </kirby-item-divider>
      <kirby-item>Item 2.1</kirby-item>
      <kirby-item>Item 2.2</kirby-item>
      <kirby-item>Item 2.3</kirby-item>
</kirby-list>
```

Or something third?

Anyways, take a look at the current section implementation and see what makes sense to reuse 😄

## Are there any additional context?
This is part of: 
- #1883 
<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by MadsBuchmann** [comment]
Oopsie duplicate of #1816
